### DISCUSSION TITLE
clear values in ui.input or ui.number by using a button

### DISCUSSION THREAD

**Discussion by minoger** [zauberzeug/nicegui#453] [Q&A] [answered]
Hello and thank you for this development.

I have a question, how to clear some inputs by using a button-element.
I have some ui.input and ui.number fields and I need a way to delete entries that have been made.

thank you 
Michael

**Comment by AL-THE-BOT-FATHER** [comment, marked as answer]
Simple example. 

```py
from nicegui import ui

def clear_number():
    number.value = None
    
def clear_input():
    input.value = None

number = ui.number()
input = ui.input()

ui.button("Clear Number", on_click=clear_number)
ui.button("Clear Input", on_click=clear_input)

ui.run()

```

**Reply by falkoschindler** [reply]
Or even shorter using a lambda statement calling `set_value()`:

```py
from nicegui import ui

number = ui.number()
input = ui.input()

ui.button("Clear Number", on_click=lambda: number.set_value(None))
ui.button("Clear Input", on_click=lambda: input.set_value(None))

ui.run()
```

**Reply by AL-THE-BOT-FATHER** [reply]
This ^^^^

**Comment by minoger** [comment]
OK - thank you - this is **very helpful**
where to find a documentation about this methods like *set_value*?

best regards
Michael

**Reply by falkoschindler** [reply]
This particular method might currently be undocumented. We are working on filling the gaps... https://github.com/zauberzeug/nicegui/issues/439

**Reply by minoger** [reply]
Thank you again. for some of my question I can find answers in the discussions ..and if no ... i will ask : - )
